File: Code/JavaScript/client/src/messages/VoltageMessage.js
export default class VoltageMessage
{
 static toJSON(o){ 
    const r = {};
   r["tpe"]="vol";
    r["v"]=o.voltage;
    r["r"]=o.rawVoltage;
    return r;
 }
 static fromJSON(o){
    const r = {};
    r.voltage=o["v"];
    r.rawVoltage=o["r"];
   return r;
 }
}

File: Code/C#/VoltageFeedbackMessages/VoltageMessage.cs
using MessageTypes.Attributes;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;
using VoltageFeedbackMessages.DataMemberNames.Requests;
using VoltageFeedbackMessages.DataMemberNames.Responses;

namespace VoltageFeedbackMessages.Messages
{
    [DataMemberNamesClass(typeof(VoltageMessageDataMemberNames))]
    public class VoltageMessage
    {
        [JsonPropertyName(VoltageMessageDataMemberNames.Voltage)]
        [JsonInclude]
        [DataMember(Name = VoltageMessageDataMemberNames.Voltage)]
        public float Voltage { get; protected set; }

        [JsonPropertyName(VoltageMessageDataMemberNames.RawVoltage)]
        [JsonInclude]
        [DataMember(Name = VoltageMessageDataMemberNames.RawVoltage)]
        public UInt16 RawVoltage { get; protected set; }
    }
}


